src/main/resources/templates/mdBenefitsFlow/choosePrograms.html

```diff
@@ -14,22 +14,29 @@
             th:replace="~{fragments/form :: form(action=${formAction}, content=~{::formContent})}">
           <th:block th:ref="formContent">
             <div class="form-card__content">
+              <h2 th:text="#{choose-programs.check-all-that-apply}"></h2>
               <th:block th:replace="~{fragments/inputs/checkboxFieldset ::
                   checkboxFieldset(inputName='programs',
-                  label=#{choose-programs.check-all-that-apply},
+                  ariaLabel=#{choose-programs.header},
                   content=~{::content})}">
                 <th:block th:ref="content">
-                  <th:block th:each="programType : ${T(org.mdbenefits.app.data.enums.ProgramType).values()}">
-                    <th:block th:if="${fieldData.get('helpNeeded[]').contains(programType.name())}">
-                      <span th:with="shortDesc=${programType.getShortDescSrc().isBlank()} ? '' : #{${programType.getShortDescSrc()}}"
-                            th:utext="${'<b>' + shortDesc + '</b>'}">
-                      </span>
+                  <th:block
+                      th:with="relevantPrograms = ${T(org.mdbenefits.app.utils.ApplicationUtilities).flattenHouseholdNeedsPrograms(fieldData.get('helpNeeded[]'))}">
+                    <th:block
+                        th:each="programType : ${T(org.mdbenefits.app.data.enums.ProgramType).values()}">
                       <th:block
-                          th:replace="~{fragments/inputs/checkboxInSet:: checkboxInSet(
+                          th:if="${relevantPrograms.contains(programType)}">
+                        <h2
+                            th:with="shortDesc=${programType.getShortDescSrc().isBlank()} ? '' : #{${programType.getShortDescSrc()}}"
+                            th:utext="${shortDesc}">
+                        </h2>
+                        <th:block class="font-weight--normal"
+                                  th:replace="~{fragments/inputs/checkboxInSet:: checkboxInSet(
```

Comment:
I don't think that putting `class="font-weight--normal"` here will do anything, as the `th:block` is a construct that's not actually put into the HTML.  The class needs to go on whatever the "checkboxInSet" produces.   You'd need to edit this code, though it already may be done (but not 100% sure it's right). https://github.com/codeforamerica/mdpilot/blob/main/src/main/resources/templates/fragments/inputs/checkboxInSet.html

Reply:
Try taking out the class stuff on this th:block and see how it renders.

Reply:
You are correct. I removed the font weight class and it didn't affect anything. Removing it.